### PR TITLE
chore: upgrade `actix-web` to `v4.0.1`

### PR DESCRIPTION
**Pull request by miraclx**
Updates;
 - `actix-web` to `v4.0.1`
 - `actix-cors` to `v0.6.1`
 - `actix-http` to `v3.0.4`
 - `paperclip` to `v0.7.0`

Fixes: https://github.com/near/nearcore/issues/6599
Fixes: https://github.com/near/nearcore/issues/6798

### CODE REVIEW

**Comment by matklad** [comment]
cargo deny fails

File: Cargo.lock
Comment (matklad) [inline Cargo.lock:3259]: Can we upgrade `strum` to avoid duplication? Might to a pre-PR for that.

File: Cargo.lock
Comment (matklad) [inline Cargo.lock:5436]: Hm, duplicating `mio` seems pretty bad, it's a core crate. Can we dedupe this as well?

**Review by matklad — APPROVED** [review]

File: Cargo.lock
Comment (miraclx) [reply Cargo.lock:3259]: Out today, but will do.

Comment (mina86) [reply Cargo.lock:3259]: FYI, I have some WIP PRs that clean up and update strum. Currently they are waiting for #6678 but I can speed them along if you want.

Comment (miraclx) [reply Cargo.lock:3259]: Would it be that jarring of a conflict if this merged first? Then again, I don't mind holding this until you're set.

Comment (mina86) [reply Cargo.lock:3259]: Nope. I’m fine either way. Updating to strum 0.24 requires some cleanups or `#[allow(deprecated)]`.

**Review by nikurt — APPROVED** [review]

File: Cargo.lock
Comment (matklad) [reply Cargo.lock:3259]: General observation: if the crate is at version `0.xx`, it's a medium evidence that using such crate means accepting a bit of churn.

Comment (mina86) [reply Cargo.lock:3259]: Actually nvm, I was being a dummy. There’s `strum::IntoStaticStr` we should be migrating to.  I’ll have a PR soon.

Comment (mina86) [reply Cargo.lock:3259]: I got strum to the state where `sed -i '/^strum/ s/0.20/0.24/' -- $(git grep -l strum)` will work for the most part. This, however, brings some issue with duplicate strum_macros dependency. This could be resolved by your change though.  Either way, I spent way too much time on this already, so I’ll let you deal with the dependency hell. ;)

**Comment by mina86** [comment]
I’ve tried a merge but we get bunch of test failures.  Someone will have to actually spend some time to get this PR over the finish line.

**Comment by miraclx** [comment]
> I’ve tried a merge but we get bunch of test failures. Someone will have to actually spend some time to get this PR over the finish line.

Yeah, had been working on this. Got stuck resolving `near-crypto` dependency version discrepancies. Still testing locally. Would update when things finally work locally.

File: Cargo.lock
Comment (miraclx) [reply Cargo.lock]: Done

File: Cargo.lock
Comment (miraclx) [reply Cargo.lock]: > Can we upgrade strum to avoid duplication? 

Done

**Comment by miraclx** [comment]
So, I've fixed all CI issues except one which I can't seem to reproduce locally.

I've retried it twice and it doesn't seem like its a fluke, but maybe it's specific to our test environment?

https://buildkite.com/nearprotocol/nearcore/builds/16246#fe8d3f24-c7a2-4881-8ccb-adf71263452b

cc: @matklad @mzhangmzz

**Comment by matklad** [comment]
Looking into this. `$ cargo test -p mock_node --lib -- test_mock_node_basic` indeed fails for me, but it takes a couple of runs to see that.

**Comment by miraclx** [comment]
Indeed, I was finally able to reproduce the error locally after **62** retries using the following command:

```console
$ while RUST_BACKTRACE=full cargo test -p mock_node -- test_mock_node_basic; do; done
```

It's a weird case, since it works so many times locally, but never in CI.

If you have any context as to why this is, please weigh in. This PR is actively blocking other tasks in the pipeline.

**Comment by matklad** [comment]
This seems to be a genuine race condition. In particular, the following diff reproduces the problem: 

```diff
diff --git a/tools/mock_node/src/setup.rs b/tools/mock_node/src/setup.rs
index 501722c07..0378cf0f9 100644
--- a/tools/mock_node/src/setup.rs
+++ b/tools/mock_node/src/setup.rs
@@ -342,6 +342,7 @@ pub fn setup_mock_node(
     #[cfg(feature = "test_features")]
     let server = None;
 
+    std::thread::sleep_ms(300);
     network_adapter.set_recipient(mock_network_actor.clone().recipient());
     (mock_network_actor, client_actor, view_client, server)
 }
```

And it also reproduces it on *master*, so the bug is pre-existing. 

However, this probably means that `start_http` which we do just above that line now takes longer with new actix, which might or might not be a problem.

**Comment by austinabell** [comment]
Noting this PR is blocking `near-vm-logic` and `near-primitives-core` releases, which in turn will delay `4.0` stable release from happening. Is there anything I can help with to bring this PR in?

Seems like the only CI failures are just due to compile errors with the migration?

**Comment by miraclx** [comment]
Thanks for the patch. @matklad

**Comment by matklad** [comment]
As this is blocking, I am going to slap auto-merge here. There's a high chance that something will come up (slow startup does seems suspecisous), but it seems easier to investigate that after the merge!

**Comment by mina86** [comment]
This broke Rosetta test: https://nayduck.near.org/#/test/320340

**Pull request merged**
